pkg/lib/journal.js

```diff
@@ -210,10 +210,11 @@ journal.printable = function printable(value) {
         return _("[no data]");
     else if (typeof (value) == "string")
         return value;
-    else if (value.length !== undefined)
-        return cockpit.format(_("[$0 bytes of binary data]"), value.length);
-    else
-        return _("[binary data]");
+    else if (value.length !== undefined && value.length <= 1000)
+        return new TextDecoder().decode(new Uint8Array(value));
+    else {
+        return _((value.length ? `[${value.length} bytes of ` : "[") + "binary data]");
```

Comment:
You can't split a translatable string that way, I'm afraid. It needs to be a single thing with a `$0` placeholder. Please just add the length check to the `if` condition, and leave the final `else` as it was.

Also, I thought we agreed on only decoding the `MESSAGE` field? The printable() function doesn't know the field name, so this needs to become a new parameter that is set by the caller, or you just pass the field name.

Reply:
ok

Reply:
because the word order can be different in different languages, or the second part gets declinated depending on the precise $0 value. Also, you want to avoid translating one half, but not the other.